Updated Documentation for EntraBeta (#837)
microsoftgraph/entra-powershell: module/docs/entra-powershell-beta/Microsoft.Graph.Entra.Beta/New-EntraBetaGroup.md

 [-Visibility <String>] 
 [-MembershipRuleProcessingState <String>]
 [-IsAssignableToRole <Boolean>] 
 [<CommonParameters>]
```

## Description

The `New-EntraBetaGroup` cmdlet creates a Microsoft Entra ID group. Specify the `DisplayName`, `MailNickname`, `MailEnabled` and `SecurityEnabled` parameters for creating a Microsoft Entra ID group.

For information about creating dynamic groups, see: [Using attributes to create advanced rules](https://learn.microsoft.com/entra/identity/users/groups-dynamic-membership).

**Notes on permissions:**

- To create the group with users as owners or members, the app must have at least the `User.Read.All` permission.
- To create the group with other service principals as owners or members, the app must have at least the `Application.Read.All` permission.
- To create the group with either users or service principals as owners or members, the app must have at least the `Directory.Read.All` permission.

## Examples

### Example 1: Create a group

```powershell
Connect-Entra -Scopes 'Group.ReadWrite.All' #Delegated Permission
Connect-Entra -Scopes 'Group.Create' #Application permission

> New-EntraBetaGroup -DisplayName 'HelpDesk admin group2' -MailEnabled $False -MailNickname 'helpDeskAdminGroup' -SecurityEnabled $True

Reviewer: ```suggestion
$params = @{
    DisplayName = 'HelpDesk admin group2'
    MailEnabled = $False
    MailNickname = 'helpDeskAdminGroup'
    SecurityEnabled = $True
}

New-EntraBetaGroup @params
```

```

```Output
DisplayName           Id                                   MailNickname       Description GroupTypes
-----------           --                                   ------------       ----------- ----------
HelpDesk admin group2 bbbbbbbb-5555-5555-0000-qqqqqqqqqqqq helpDeskAdminGroup             {}
```

This example demonstrates how to create the new group.

### Example 2: Create a group with Description parameter

```powershell
Connect-Entra -Scopes 'Group.ReadWrite.All' #Delegated Permission
Connect-Entra -Scopes 'Group.Create' #Application permission

> New-EntraBetaGroup -DisplayName 'HelpDesk admin group' -MailEnabled $false -MailNickname 'helpDeskAdminGroup' -SecurityEnabled $true  -Description 'Group assignable to role'

Reviewer: ```suggestion
$params = @{
    DisplayName = 'HelpDesk admin group'
    MailEnabled = $false
    MailNickname = 'helpDeskAdminGroup'
    SecurityEnabled = $true
    Description = 'Group assignable to role'
}

New-EntraBetaGroup @params
```

```

```Output

DisplayName          Id                                   MailNickname       Description              GroupTypes
-----------          --                                   ------------       -----------              ----------
HelpDesk admin group zzzzzzzz-6666-8888-9999-pppppppppppp helpDeskAdminGroup Group assignable to role {}

```

This example demonstrates how to create the new group with description parameter.

### Example 3: Create a group with IsAssignableToRole parameter

```powershell
Connect-Entra -Scopes 'Group.ReadWrite.All' #Delegated Permission
Connect-Entra -Scopes 'Group.Create' #Application permission

> New-EntraBetaGroup -DisplayName 'HelpDesk admin group2' -Description 'Group assignable to role' -MailEnabled $False -MailNickname 'helpDeskAdminGroup' -SecurityEnabled $True -IsAssignableToRole $True 

Reviewer: ```suggestion
$params = @{
    DisplayName = 'HelpDesk admin group2'
    Description = 'Group assignable to role'
    MailEnabled = $False
    MailNickname = 'helpDeskAdminGroup'
    SecurityEnabled = $True
    IsAssignableToRole = $True
}

New-EntraBetaGroup @params
```

```

```Output
DisplayName           Id                                   MailNickname       Description              GroupTypes
-----------           --                                   ------------       -----------              ----------
HelpDesk admin group2 vvvvvvvv-8888-9999-0000-jjjjjjjjjjjj helpDeskAdminGroup Group assignable to role {}
```

This example demonstrates how to create the new group with IsAssignableToRole parameter.

### Example 4: Create a group with Visibility parameter

```powershell
Connect-Entra -Scopes 'Group.ReadWrite.All' #Delegated Permission
Connect-Entra -Scopes 'Group.Create' #Application Permission

> New-EntraBetaGroup -DisplayName 'HelpDesk admin group2' -Description 'Group assignable to role' -MailEnabled $False -MailNickname 'helpDeskAdminGroup' -SecurityEnabled $True -Visibility 'Private'

Reviewer: ```suggestion
$params = @{
    DisplayName = 'HelpDesk admin group2'
    Description = 'Group assignable to role'
    MailEnabled = $False
    MailNickname = 'helpDeskAdminGroup'
    SecurityEnabled = $True
    Visibility = 'Private'
}

New-EntraBetaGroup @params
```

```

```Output
DisplayName           Id                                   MailNickname       Description              GroupTypes
-----------           --                                   ------------       -----------              ----------
HelpDesk admin group2 gggggggg-0000-4444-3333-llllllllllll helpDeskAdminGroup Group assignable to role {}
```

This example demonstrates how to create the new group with Visibility parameter.

### Example 5: Create a group with GroupTypes parameter

```powershell
Connect-Entra -Scopes 'Group.ReadWrite.All' #Delegated Permission
Connect-Entra -Scopes 'Group.Create' #Application permission

> New-EntraBetaGroup -DisplayName 'HelpDesk admin group3' -Description 'group des' -MailEnabled $False -MailNickname 'helpDeskAdminGroup1' -SecurityEnabled $True -GroupTypes 'Unified'

Reviewer: ```suggestion
$params = @{
    DisplayName = 'HelpDesk admin group3'
    Description = 'group des'
    MailEnabled = $False
    MailNickname = 'helpDeskAdminGroup1'
    SecurityEnabled = $True
    GroupTypes = 'Unified'
}

New-EntraBetaGroup @params
```

```

```Output
DisplayName           Id                                   MailNickname        Description GroupTypes
-----------           --                                   ------------        ----------- ----------
HelpDesk admin group3 xxxxxxxx-8888-5555-9999-bbbbbbbbbbbb helpDeskAdminGroup1 group des   {Unified}
```

This example demonstrates how to create the new group with GroupTypes parameter.

### Example 6: Create a group membership rule processing state parameter

```powershell
Connect-Entra -Scopes 'Group.ReadWrite.All' #Delegated Permission
Connect-Entra -Scopes 'Group.Create' #Application permission

> New-EntraBetaGroup -DisplayName 'HelpDesk admin group2'  -MailEnabled $False -MailNickname 'helpDeskAdminGroup' -SecurityEnabled $True  -MembershipRuleProcessingState 'On'

Reviewer: ```suggestion
$params = @{
    DisplayName = 'HelpDesk admin group2'
    MailEnabled = $False
    MailNickname = 'helpDeskAdminGroup'
    SecurityEnabled = $True
    MembershipRuleProcessingState = 'On'
}

New-EntraBetaGroup @params
```

```

```Output
DisplayName           Id                                   MailNickname       Description GroupTypes
-----------           --                                   ------------       ----------- ----------
HelpDesk admin group2 xxxxxxxx-8888-5555-9999-bbbbbbbbbbbb helpDeskAdminGroup             {}
```

This example demonstrates how to create the new group with MembershipRuleProcessingState parameter

### Example 7: Create a group membership rule parameter

```powershell
Connect-Entra -Scopes 'Group.ReadWrite.All' #Delegated Permission
Connect-Entra -Scopes 'Group.Create' #Application permission

> New-EntraBetaGroup -DisplayName 'HelpDesk admin group2'  -MailEnabled $False -MailNickname 'helpDeskAdminGroup' -SecurityEnabled $True -MembershipRule '(user.department -contains "Marketing")' -MembershipRuleProcessingState 'On'

Reviewer: ```suggestion
$params = @{
    DisplayName = 'HelpDesk admin group2'
    MailEnabled = $False
    MailNickname = 'helpDeskAdminGroup'
    SecurityEnabled = $True
    MembershipRule = '(user.department -contains "Marketing")'
    MembershipRuleProcessingState = 'On'
}

New-EntraBetaGroup @params
```

```

### -MembershipRule

Specifies the membership rule for a dynamic group.

> For more information about the rules that you can use for dynamic groups, see Using attributes to create advanced rules (https://azure.microsoft.com/documentation/articles/active-directory-accessmanagement-groups-with-advanced-rules/).

Reviewer: ```suggestion
For more information about the rules that you can use for dynamic groups, see Using attributes to create advanced rules (https://learn.microsoft.com/entra/identity/users/groups-dynamic-membership).
```

```

### CommonParameters

This cmdlet supports the common parameters: `-Debug`, `-ErrorAction`, `-ErrorVariable`, `-InformationAction`, `-InformationVariable`, `-OutVariable`, `-OutBuffer`, `-PipelineVariable`, `-Verbose`, `-WarningAction`, and `-WarningVariable`. For more information, see [about_CommonParameters](https://go.microsoft.com/fwlink/?LinkID=113216).

## Inputs

### None

## Outputs

### System.Object

## Notes

This cmdlet is currently in Public Preview.
While a cmdlet is in Public Preview, we may make changes to the cmdlet, which could have unexpected effects.
We recommend that you don't use this cmdlet in a production environment.

## Related links

[Get-EntraBetaGroup](Get-EntraBetaGroup.md)

[Remove-EntraBetaGroup](Remove-EntraBetaGroup.md)

[Set-EntraBetaGroup](Set-EntraBetaGroup.md)

> [Using attributes to create advanced rules](https://azure.microsoft.com/documentation/articles/active-directory-accessmanagement-groups-with-advanced-rules/)

Reviewer: ```suggestion
[Using attributes to create advanced rules](https://learn.microsoft.com/entra/identity/users/groups-dynamic-membership)
```
